packages/contracts-router/contracts/test/Router.t.sol

```diff
@@ -0,0 +1,50 @@
+// SPDX-License-Identifier: MIT OR Apache-2.0
+pragma solidity 0.7.6;
+
+import "forge-std/Test.sol";
+import {RouterHarness} from "./harnesses/RouterHarness.sol";
+
+contract RouterTest is Test {
+    RouterHarness router;
+
+    uint32 testDomain;
+    bytes32 testRouter;
+
+    function setUp() public {
+        testDomain = 123213;
+        testRouter = "router address";
+        router = new RouterHarness();
+        address xAppConnectionManager = vm.addr(1321);
+        router.exposed__XAppConnectionClient_initialize(xAppConnectionManager);
+        router.enrollRemoteRouter(testDomain, testRouter);
+    }
+
+    function test_enrollRemoteRouter() public {
+        bytes32 storedRouter = router.remotes(testDomain);
```

Comment:
wdyt about 
```
        assertEq(router.remotes(testDomain), address(0));
        router.enrollRemoteRouter(testDomain, testRouter);
        assertEq(router.remotes(testDomain), testRouter);
```
test the state before & after

Reply:
you could also add all of the before / after checks here

```
        // domain + address is NOT enrolled
        assertEq(router.remotes(testDomain), address(0));
        assert(!router.exposed_isRemoteRouter(testDomain, testRouter));
        vm.expectRevert("!remote");
        router.exposed_mustHaveRemote(testDomain);
        // enroll router
        router.enrollRemoteRouter(testDomain, testRouter);
        // domain + address is enrolled
        assertEq(router.remotes(testDomain), testRouter);
        assert(router.exposed_isRemoteRouter(testDomain, testRouter));
        assertEq(router.exposed_mustHaveRemote(testDomain), testRouter);
```

Reply:
alternatively, you could add the before / after checks to each individual test case (i think this might be the cleaner option). move `enrollRemoteRouter` from the setup function and place it as a helper function, and use it before / after in each test case